### DISCUSSION TITLE
Spectral environment maps

### DISCUSSION THREAD

**Discussion by jussihh** [mitsuba-renderer/mitsuba3#986] [Q&A] [answered]
Is there support for spectral environment maps? E.g. implementing support for EXR files in Fichet et al. layout would be great.

Alban Fichet, Romain Pacanowski, and Alexander Wilkie, An OpenEXR Layout for Spectral Images, Journal of Computer Graphics Techniques (JCGT), vol. 10, no. 3, 1-18, 2021
Available online [http://jcgt.org/published/0010/03/01/]

**Comment by njroussel** [comment, marked as answer]
Hi @jussihh 

Our `envmap` is compatible with `spectral` variants. But it does expect the input `Bitmap` object to use our [spectral upsampling representation](https://rgl.epfl.ch/publications/Jakob2019Spectral).
So you would either need to convert your EXR file to this representation or add support in the `envmap` for another type of spectral representations.
